lib/AppInfo/Application.php

```diff
@@ -101,6 +101,8 @@ public function registerFilesActivity() {
 		Util::connectHook('\OCA\Files_Trashbin\Trashbin', 'post_restore', 'OCA\Activity\FilesHooksStatic', 'fileRestore');
 		Util::connectHook('OCP\Share', 'post_shared', 'OCA\Activity\FilesHooksStatic', 'share');
 		Util::connectHook('OCP\Share', 'pre_unshare', 'OCA\Activity\FilesHooksStatic', 'unShare');
+		Util::connectHook('\OC\Files\Storage\Shared', 'fopen', 'OCA\Activity\FilesHooksStatic', 'downloadedShare');
+		Util::connectHook('\OC\Files\Storage\Shared', 'file_get_contents', 'OCA\Activity\FilesHooksStatic', 'downloadedShare');
```

Comment:
This is for both. I've only used it with local users, so didn't experience the missing events, but it would be good to make it work with remote shares as well. I had a quick look at the code and think the storage wrapper is where there should be something of interest for this, since this is what the webdav endpoint uses:
https://github.com/nextcloud/server/blob/638f852ed85af759a0ce71397a1f23080c5cf14e/apps/dav/appinfo/v1/publicwebdav.php